src/gtfs.py

```diff
@@ -209,3 +252,42 @@ def add_gtfs_headways(events_df: pd.DataFrame, all_trips: pd.DataFrame, all_stop
         results.append(augmented_events)
 
     return pd.concat(results)
+
+
+current_gtfs_archive = None
+write_gtfs_archive_lock = Lock()
+
+
+def update_current_gtfs_archive_if_necessary():
+    global current_gtfs_archive
+    global write_gtfs_archive_lock
+    with write_gtfs_archive_lock:
+        gtfs_service_date = util.service_date(datetime.datetime.now(util.EASTERN_TIME))
+        needs_update = current_gtfs_archive is None or current_gtfs_archive.service_date != gtfs_service_date
+        if needs_update:
+            if current_gtfs_archive is None:
+                logger.info(f"Downloading GTFS archive for {gtfs_service_date}")
+            else:
+                logger.info(f"Updating GTFS archive from {current_gtfs_archive.service_date} to {gtfs_service_date}")
+            current_gtfs_archive = read_gtfs(gtfs_service_date)
```

Comment:
can you merge `main` and use the `routes_filter` arg here? it wont speed anything up, but itll reduce the size of the archive we keep in memory globally

Reply:
Have a look, I passed `ALL_ROUTES` in as the filter now.